### PR TITLE
Make SessionToken on QueryRequestOptions public

### LINKED ISSUE #973: Retrieve multiple documents with Session Token
**Is your feature request related to a problem? Please describe.**
When using GetItemQueryIterator I cannot see a way to pass in the session token.

**Describe the solution you'd like**
A way to retrieve multiple documents passing in a session token. One way would be to add session token to QueryRequestOptions and use it.

**Describe alternatives you've considered**
I can retrieve a single item using ReadItemAsync and pass in the session token as part of the options.

**Additional context**
Maybe there is a way to do this, and I've just not found it? I also opened a Stack Overflow question a few days ago, but had no response on that yet https://stackoverflow.com/questions/58644522/net-core-cosmosdb-get-multiple-documents-with-session

### PR DESCRIPTION
**Pull request by dpgregory**
## Description

Made SessionToken public on QueryRequestOptions to allow querying of multiple documents with SessionToken.

## Type of change
- [ ] New feature (non-breaking change which adds functionality)

## Closing issues

Closes #973

### CODE REVIEW

File: Microsoft.Azure.Cosmos/src/RequestOptions/QueryRequestOptions.cs [hunk 1]
```diff
@@ -138,7 +138,7 @@ public class QueryRequestOptions : RequestOptions
         ///
         /// </para>
         /// </remarks>
-        internal string SessionToken { get; set; }
+        public string SessionToken { get; set; }
```
Comment (j82w) [inline Microsoft.Azure.Cosmos/src/RequestOptions/QueryRequestOptions.cs]: Please add a note to the changelog

Comment (dpgregory) [reply Microsoft.Azure.Cosmos/src/RequestOptions/QueryRequestOptions.cs]: I think I've done this now, let me know if it's not correct

File: changelog.md [hunk 2]
```diff
@@ -9,6 +9,10 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## <a name="3.4.1"/> [3.4.1](https://www.nuget.org/packages/Microsoft.Azure.Cosmos/3.4.1) - 2019-11-06
 
+### Added
+
+- [#853](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/979) Make SessionToken on QueryRequestOptions public.
```
Comment (j82w) [inline changelog.md]: Please move this above 3.4.1 under the Unreleased tag. 3.4.1 is already released

Comment (j82w) [reply changelog.md]: Sorry for all the azp comments. It turns out that Github had a bug that prevented the gates from running on forked repos.

Comment (dpgregory) [reply changelog.md]: No worries. Sorry, I think I've got it in the right place now. This is my first contribution to open source project, so it's a learning experience. I thought it best to try and fix my own issue though.

**Comment by dpgregory** [comment]
Is there anything else I need to do on this, or is it just waiting for someone to be free to look at it?

**Comment by kirankumarkolli** [comment]
@bchong95 , @sboshra please take a look. 

Exposing this contract enables read-my-writes through queries as well.

File: Microsoft.Azure.Cosmos/src/RequestOptions/QueryRequestOptions.cs (hunk 1 above)
Comment (bchong95) [reply Microsoft.Azure.Cosmos/src/RequestOptions/QueryRequestOptions.cs]: Even though this is made public we need to wire it through the query stack right?

Comment (dpgregory) [reply Microsoft.Azure.Cosmos/src/RequestOptions/QueryRequestOptions.cs]: From my understanding this is already done on the line 164:

RequestOptions.SetSessionToken(request, this.SessionToken);

Comment (bchong95) [reply Microsoft.Azure.Cosmos/src/RequestOptions/QueryRequestOptions.cs]: Good point. I just want to make sure what the story will be. If a user does multiple point writes, then what session token will they send for the query?

Comment (kirankumarkolli) [reply Microsoft.Azure.Cosmos/src/RequestOptions/QueryRequestOptions.cs]: That's a good point but also a different question. 
SessionToken would be same as what is used for any ReadItemAsync.

Comment (dpgregory) [reply Microsoft.Azure.Cosmos/src/RequestOptions/QueryRequestOptions.cs]: Is there anything else I need to change, or are we all good now?

Comment (kirankumarkolli) [reply Microsoft.Azure.Cosmos/src/RequestOptions/QueryRequestOptions.cs]: Nope we are good to ship it. Thanks @dpgregory for your contributions.

Comment (dpgregory) [reply Microsoft.Azure.Cosmos/src/RequestOptions/QueryRequestOptions.cs]: Great! No worries, it wasn't much!

**Pull request merged**